### DISCUSSION TITLE
What does the proxy / api do?

### DISCUSSION THREAD

**Discussion by royduin** [searchkit/searchkit#1273] [Q&A] [answered]
The docs say is recommended to use a proxy in production and the examples refer to `@searchkit/api`, but what does it do? Just proxy? That can also be archived with a simple Nginx config and Elasticsearch can be secured with xpack; for example: https://docs.rapidez.io/1.x/deployment.html#secure-elasticsearch, is there any difference?

**Comment by joemcelroy** [comment]
Hey there! Sorry for late reply.

Rather than all the work of transforming ui search requests (query, filters, sorting) into elasticsearch queries and performing network requests to elasticsearch done in the browser, all of this can be done in node. 

When configured in this way using searchkit/api, Searchkit instantsearch client will forward the ui search requests to your api server instead.

this is great for those who don’t want to expose their elasticsearch host publicly and provide a layer in between the browser and server with performance monitoring, connection throttling.

**Comment by joemcelroy** [comment, marked as answer]
The difference between the link you shared and searchkit/api is that whilst elasticsearch is read only, you can still perform any query you want to elasticsearch, potentially raising DOS attacks with crafting expensive queries. 

Searchkit api hides elasticsearch and provides an api built for search use-cases. The elasticsearch queries are generated on the node api rather than on the browser.

**Comment by royduin** [comment]
So when Searchkit communicates directly with Elasticsearch it produces JQL queries? When `@searchkit/api` is used a custom API is used and `@searchkit/api` transforms the requests to JQL?

Performance monitoring, connection throttling, etc could be configured but `@searchkit/api` just transforms requests?

**Reply by joemcelroy** [reply]
Yep. 

Another advantage is that your node API could work with a session to provide base filters to the request. For example you only want to search documents linked to the customer's role.

**Reply by royduin** [reply]
Thank you for the clarification!

**Reply by joemcelroy** [reply]
and also just to be super clear, Searchkit can work the same way as the approach you shared in your link if you dont have the ability to put a node layer in between. it just unlocks a few extra features that is an advantage over say Reactivesearch.

**Reply by royduin** [reply]
Clearly, thanks for your time!
